x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/ApiKeyService.java

```diff
@@ -1145,6 +1155,8 @@ private boolean verify(SecureString password) {
         final BytesReference roleDescriptorsBytes;
         final BytesReference limitedByRoleDescriptorsBytes;
         final Map<String, Object> creator;
+        @Nullable
+        final BytesReference metadataFlattened;
```

Comment:
Why is this tracked as bytes rather than a Map?

Reply:
This is to avoid unnecessary de/serialisation cost similar to the role descriptors, especially since the metadata is not in use at all (not until we expose it in some ingest processor).